src/test/java/org/littleshoot/proxy/DirectRequestTest.java

```diff
@@ -0,0 +1,108 @@
+package org.littleshoot.proxy;
+
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.Assert.assertThat;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+
+import java.net.Socket;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.littleshoot.proxy.impl.DefaultHttpProxyServer;
+import org.littleshoot.proxy.test.SocketClientUtil;
+
+/**
+ * This class tests direct requests to the proxy server, which causes endless
+ * loops (#205).
+ */
+public class DirectRequestTest {
+    private HttpProxyServer proxyServer;
+
+    private Socket socket;
+
+    @Before
+    public void setUp() throws Exception {
+        socket = null;
+        proxyServer = null;
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        try {
+            if (proxyServer != null) {
+                proxyServer.abort();
+            }
+        } finally {
+            if (socket != null) {
+                socket.close();
+            }
+        }
+    }
+
+    @Test(timeout = 5000)
+    public void testDirectRequestAnsweredBadRequest() throws Exception {
+
+        HttpFiltersSource filtersSource = new HttpFiltersSourceAdapter() {
+            @Override
+            public HttpFilters filterRequest(HttpRequest originalRequest) {
+                return new HttpFiltersAdapter(originalRequest);
+            }
+        };
+
+        proxyServer = DefaultHttpProxyServer.bootstrap()//
+                .withPort(0)//
+                .withFiltersSource(filtersSource)//
+                .start();
+
+        socket = SocketClientUtil.getSocketToProxyServer(proxyServer);
```

Comment:
It would be nice to see an HTTPS test case as well (which would be far easier with Apache HTTP client).